### PR TITLE
Solve the error when try to remove a file previously removed, the idea is to remove first the imagekeys files and then, the remaining files of the temporary directory

### PR DESCRIPTION
**Pull request by josepsanzcamp**
Solve the error when try to remove a file previously removed, the idea is to remove first the imagekeys files and then, the remaining files of the temporary directory

### CODE REVIEW

**Comment by williamdes** [comment]
Would you mind adding an example that covers your case file to `examples/` ?

**Comment by josepsanzcamp** [comment]
> Would you mind adding an example that covers your case file to `examples/` ?

Hi @williamdes, currently my aportation only optimize the execution, currently not solves any error because I see that the latest code protect the original error by the file_exists, but in essency, the idea is to first remove the knowed files and then, search for untracked files to remove too.

This patch was originaly PR in the TCPDF repository (https://github.com/tecnickcom/TCPDF/pull/340) but someone comments me that maybe is more better to PR to this repository instead the abandonated TCPDF repository.

In any case, the title or the PR is not correct, because with the 7.0.0 the error was fixed by the file_exists but the code movement that I propose is more optimal because unlink the knowed files and then unlink the untracked _tcpdf_ files.

Thanks in advance by your maintenance and evolution work in this library.

**Comment by williamdes** [comment]
> Thanks in advance by your maintenance and evolution work in this library.

Thank you too for contributing

**Review by williamdes — APPROVED** [review]
Good improvement !

File: tcpdf.php [hunk 1]
```diff
@@ -7710,7 +7710,15 @@ public function _destroy($destroyall=false, $preserve_objcopy=false) {
 		}
 		if ($destroyall AND !$preserve_objcopy) {
 			self::$cleaned_ids[$this->file_id] = true;
-			// remove all temporary files
+			// first: remove all imagekeys files
+			if (isset($this->imagekeys)) {
+				foreach($this->imagekeys as $file) {
+					if (strpos($file, K_PATH_CACHE) === 0 && file_exists($file)) {
+						unlink($file);
```
Comment (williamdes) [inline tcpdf.php:7717]: Note: the STFU operator was removed, I assume it was intentional

Comment (josepsanzcamp) [reply tcpdf.php:7717]: Yes, I understand that was used initially to hide the original error when try to unlink an unlinked previously file

**Pull request merged**